core/src/env/srml/srml_only/impls.rs

```diff
@@ -72,6 +72,18 @@ impl EnvStorage for SrmlEnvStorage {
     }
 }
 
+/// Wrapped access to return data.
+/// get() Must be used right after ReturnData is returned.
+#[allow(unused)]
+pub struct ReturnData {}
+
+#[allow(unused)]
+impl ReturnData {
+    fn get() -> Vec<u8> {
```

Comment:
The only thing now missing is making `ReturnData` generic over a `T: Decode` so that `get` returns a `T` directly `fn get() -> T`. It should panic (via `Result::expect`) if the decoding failed. For convenience feel free to also add a `fn get_raw() -> Vec<u8>` that simply does what the current `get` does.

Reply:
Also `get` should be public.